commit a12982cf2b5fd1975ff58a877d58a54a5c1b3a7d
Author: wolfgang <github@linux-dude.de>
Date:   Fri Apr 9 12:59:41 2021 +0200

    compile patch for gentoo

diff --git a/.cargo/config b/.cargo/config
index 3b5b6e48..93dc0f94 100644
--- a/.cargo/config
+++ b/.cargo/config
@@ -1,5 +1,18 @@
-[source]
-[source.debian-packages]
-directory = "/usr/share/cargo/registry"
 [source.crates-io]
-replace-with = "debian-packages"
+replace-with = "vendored-sources"
+
+[source."git://git.proxmox.com/git/pathpatterns.git"]
+git = "git://git.proxmox.com/git/pathpatterns.git"
+replace-with = "vendored-sources"
+
+[source."git://git.proxmox.com/git/proxmox-fuse.git"]
+git = "git://git.proxmox.com/git/proxmox-fuse.git"
+replace-with = "vendored-sources"
+
+[source."git://git.proxmox.com/git/pxar.git"]
+git = "git://git.proxmox.com/git/pxar.git"
+replace-with = "vendored-sources"
+
+[source.vendored-sources]
+directory = "../cargo_home/gentoo"
+
diff --git a/Cargo.toml b/Cargo.toml
index 97aa79f2..0a70adf9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,6 @@ name = "proxmox_backup"
 path = "src/lib.rs"
 
 [dependencies]
-apt-pkg-native = "0.3.2"
 base64 = "0.12"
 bitflags = "1.2.1"
 bytes = "1.0"
@@ -47,13 +46,15 @@ pam-sys = "0.5"
 percent-encoding = "2.1"
 pin-utils = "0.1.0"
 pin-project = "1.0"
-pathpatterns = "0.1.2"
-proxmox = { version = "0.11.0", features = [ "sortable-macro", "api-macro", "websocket" ] }
+#pathpatterns = "0.1.2"
+pathpatterns = { path = "../pathpatterns" }
+#proxmox = { version = "0.11.0", features = [ "sortable-macro", "api-macro", "websocket" ] }
 #proxmox = { git = "git://git.proxmox.com/git/proxmox", version = "0.1.2", features = [ "sortable-macro", "api-macro" ] }
-#proxmox = { path = "../proxmox/proxmox", features = [ "sortable-macro", "api-macro", "websocket" ] }
-proxmox-fuse = "0.1.1"
-pxar = { version = "0.10.1", features = [ "tokio-io" ] }
-#pxar = { path = "../pxar", features = [ "tokio-io" ] }
+proxmox = { path = "../proxmox/proxmox", features = [ "sortable-macro", "api-macro", "websocket" ] }
+#proxmox-fuse = "0.1.1"
+proxmox-fuse = { path = "../proxmox-fuse" }
+#pxar = { version = "0.10.1", features = [ "tokio-io" ] }
+pxar = { path = "../pxar", features = [ "tokio-io" ] }
 regex = "1.2"
 rustyline = "7"
 serde = { version = "1.0", features = ["derive"] }
diff --git a/src/api2/node.rs b/src/api2/node.rs
index 1f3e46a9..ce4e449f 100644
--- a/src/api2/node.rs
+++ b/src/api2/node.rs
@@ -26,7 +26,6 @@ use crate::server::WorkerTask;
 use crate::tools;
 use crate::tools::ticket::{self, Empty, Ticket};
 
-pub mod apt;
 pub mod disks;
 pub mod dns;
 pub mod network;
@@ -313,7 +312,6 @@ fn upgrade_to_websocket(
 }
 
 pub const SUBDIRS: SubdirMap = &[
-    ("apt", &apt::ROUTER),
     ("disks", &disks::ROUTER),
     ("dns", &dns::ROUTER),
     ("journal", &journal::ROUTER),
diff --git a/src/tools.rs b/src/tools.rs
index 7e3bff7b..78fa7d47 100644
--- a/src/tools.rs
+++ b/src/tools.rs
@@ -18,7 +18,6 @@ use percent_encoding::{utf8_percent_encode, AsciiSet};
 pub use proxmox::tools::fd::Fd;
 
 pub mod acl;
-pub mod apt;
 pub mod async_io;
 pub mod borrow;
 pub mod cert;
diff --git a/src/tools/daemon.rs b/src/tools/daemon.rs
index d298bf16..73699f8e 100644
--- a/src/tools/daemon.rs
+++ b/src/tools/daemon.rs
@@ -19,7 +19,6 @@ use proxmox::tools::io::{ReadExt, WriteExt};
 use crate::server;
 use crate::tools::{fd_change_cloexec, self};
 
-#[link(name = "systemd")]
 extern "C" {
     fn sd_journal_stream_fd(identifier: *const c_uchar, priority: c_int, level_prefix: c_int) -> c_int;
 }
@@ -346,7 +345,6 @@ async fn wait_service_is_not_state(service: &str, state: &str) -> Result<(), Err
     Ok(())
 }
 
-#[link(name = "systemd")]
 extern "C" {
     fn sd_notify(unset_environment: c_int, state: *const c_char) -> c_int;
 }
